src/receive_imf.rs

```diff
@@ -1800,46 +1791,76 @@ async fn apply_group_changes(
         }
     }
 
-    // Recreate the member list.
-    if recreate_member_list {
-        // Only delete old contacts if the sender is not a classical MUA user:
-        // Classical MUA users usually don't intend to remove users from an email
-        // thread, so if they removed a recipient then it was probably by accident.
-        if mime_parser.has_chat_version() {
-            context
-                .sql
-                .execute("DELETE FROM chats_contacts WHERE chat_id=?;", (chat_id,))
-                .await?;
-        }
-
-        let mut members_to_add = HashSet::new();
-        members_to_add.extend(to_ids);
-        members_to_add.insert(ContactId::SELF);
-
+    if allow_member_list_changes {
+        let mut new_members = HashSet::new();
+        new_members.extend(to_ids);
+        new_members.insert(ContactId::SELF);
         if !from_id.is_special() {
-            members_to_add.insert(from_id);
+            new_members.insert(from_id);
+        }
+
+        if !recreate_member_list {
+            let diff: HashSet<ContactId> =
+                chat_contacts.difference(&new_members).copied().collect();
+            // Only delete old contacts if the sender is not a classical MUA user:
+            // Classical MUA users usually don't intend to remove users from an email
+            // thread, so if they removed a recipient then it was probably by accident.
+            if mime_parser.has_chat_version() {
+                // This is what provides group membership consistency: we prefer removing members
```

Comment:
So, we can discuss whether this has sense, but otherwise the commit looks useful for me. If we decide to allow groups to diverge for members, the commit should be renamed though